pennylane/pulse/parametrized_hamiltonian.py

```diff
@@ -259,17 +260,26 @@ def __repr__(self):
         terms = []
 
         for coeff, op in zip(self.coeffs_fixed, self.ops_fixed):
-            term = f"({coeff}*({op}))"
+            term = f"{coeff} * {op}"
             terms.append(term)
 
         for i, (coeff, op) in enumerate(zip(self.coeffs_parametrized, self.ops_parametrized)):
             if callable(coeff) and hasattr(coeff, "__name__"):
-                term = f"({coeff.__name__}(params_{i}, t)*({op}))"
+                term = (
+                    f"{coeff.__name__}(params_{i}, t) * ({op})"
+                    if isinstance(op, Sum)
+                    else f"{coeff.__name__}(params_{i}, t) * {op}"
+                )
             elif hasattr(coeff, "__class__") and hasattr(coeff.__class__, "__name__"):
-                term = f"({coeff.__class__.__name__}(params_{i}, t)*({op}))"
+                term = (
+                    f"{coeff.__class__.__name__}(params_{i}, t) * ({op})"
+                    if isinstance(op, Sum)
+                    else f"{coeff.__class__.__name__}(params_{i}, t) * {op}"
```

Comment:
not really from this PR, I guess, but how confident are we that if the `coeff` doesn't have a name, its class will have a name? Do all classes have names?

Reply:
I think it defaults to the actual name of the class description
```python
class asd:
    pass

a = asd()
print(a.__class__.__name__)
```

```
'asd'
```